queue_item.go

```diff
@@ -0,0 +1,137 @@
+	SecondsInPreviousAttempts          uint                   `json:"SecondsInPreviousAttempts,omitempty"`
+	AncestorID                         *uint                  `json:"AncestorId,omitempty"`
+	RetryNumber                        uint                   `json:"RetryNumber,omitempty"`
+	SpecificData                       string                 `json:"SpecificData,omitempty"`
+	CreationTime                       string                 `json:"CreationTime,omitempty"`
+	Progress                           string                 `json:"Progress,omitempty"`
+	RowVersion                         string                 `json:"RowVersion,omitempty"`
+	OrganizationUnitID                 uint                   `json:"OrganizationUnitId,omitempty"`
+	OrganizationUnitFullyQualifiedName string                 `json:"OrganizationUnitFullyQualifiedName,omitempty"`
+	ProcessingException                *ProcessingException   `json:"ProcessingException,omitempty"`
+	SpecificContent                    map[string]interface{} `json:"SpecificContent,omitempty"`
+	Output                             map[string]interface{} `json:"Output,omitempty"`
+	Name                               string                 `json:"Name"`
+}
+
+// QueueItemList defines what the queue item list model looks like
+type QueueItemList struct {
+	Count int         `json:"@odata.count"`
+	Value []QueueItem `json:"value"`
+}
+
+// QueueItemCreateRequest defines how the request looks like when creating a queue item
+type QueueItemCreateRequest struct {
+	ItemData QueueItem `json:"itemData"`
+}
+
+// ProcessingException defines the structure of the queue item exception
+type ProcessingException struct {
+	Reason                  string
+	Details                 string
+	Type                    string
+	AssociatedImageFilePath string
+	CreationTime            time.Time
+}
+
+// Store creates and stores a queue item in the uipath orchestrator
+func (q *QueueItemHandler) Store(queueItem QueueItem) (QueueItem, error) {
+	var result QueueItem
+
+	queueItemCreateRequest := QueueItemCreateRequest{
+		ItemData: queueItem,
+	}
+
+	url := fmt.Sprintf("%s%s", q.Client.URLEndpoint, "Queues/UiPathODataSvc.AddQueueItem")
+
+	resp, err := q.Client.SendWithAuthorization("POST", url, queueItemCreateRequest, q.buildHeaders(), map[string]string{})
+	if err != nil {
+		return result, err
+	}
+
+	if err = json.Unmarshal(resp, &result); err != nil {
+		return result, err
+	}
+
+	return result, nil
+}
+
+// GetByID fetches a queue item by id
+func (q *QueueItemHandler) GetByID(ID uint) (QueueItem, error) {
+	var queueItem QueueItem
+
+	url := fmt.Sprintf("%s%s(%d)", q.Client.URLEndpoint, "QueueItems", ID)
+
+	resp, err := q.Client.SendWithAuthorization("GET", url, nil, q.buildHeaders(), map[string]string{})
+	if err != nil {
+		return queueItem, err
+	}
+
+	if err = json.Unmarshal(resp, &queueItem); err != nil {
+		return queueItem, err
+	}
+
+	return queueItem, nil
```

Comment:
We can simplify this error handling.

```suggestion
	err = json.Unmarshal(resp, &queueItem)

	return queueItem, err
```